src/serac/physics/tests/test_solid_functional_statics.cpp

```diff
@@ -0,0 +1,290 @@
+// Copyright (c) 2019-2022, Lawrence Livermore National Security, LLC and
+// other Serac Project Developers. See the top-level LICENSE file for
+// details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#include "serac/physics/solid_functional.hpp"
+
+#include <functional>
+#include <set>
+#include <string>
+
+#include "axom/slic/core/SimpleLogger.hpp"
+#include <gtest/gtest.h>
+#include "mfem.hpp"
+
+#include "serac/mesh/mesh_utils.hpp"
+#include "serac/physics/state/state_manager.hpp"
+#include "serac/physics/materials/solid_functional_material.hpp"
+#include "serac/serac_config.hpp"
+
+namespace serac {
+
+using solid_mechanics::direct_static_options;
+
+/**
+ * @brief Exact displacement solution that is an affine function
+ *
+ * @tparam dim number of spatial dimensions
+ */
+template <int dim>
+class AffineSolution {
+ public:
+  AffineSolution():
+    A(dim), b(dim)
+  {
+    // clang-format off
+    A(0, 0) = 0.110791568544027; A(0, 1) = 0.230421268325901;
+    A(1, 0) = 0.198344644470483; A(1, 1) = 0.060514559793513;
+    if constexpr (dim == 3) {
+      A(0, 2) = 0.15167673653354;
+      A(1, 2) = 0.084137393813728;
+      A(2, 0) = 0.011544253485023; A(2, 1) = 0.060942846497753; A(2, 2) = 0.186383473579596;
```

Comment:
I think I see the point of this formatting, to make the code graphically represent the matrix terms that are affected. It could also be possible to do something like:
```suggestion
                                                                A(0, 2) = 0.15167673653354;
                                                                A(1, 2) = 0.084137393813728;
      A(2, 0) = 0.011544253485023; A(2, 1) = 0.060942846497753; A(2, 2) = 0.186383473579596;
```

Also, if the values of these entries is not important, then it's possible to make this initialization a little more succinct (using a different arbitrary choice of values)
```cpp
for (int i = 0; i < dim; i++) {
  for(int j = 0; j < dim; j++) {
    A(i,j) = 1.0 / (i + 2 * j);
  }
  b[i] = sin(i);
}
```
would have a similar effect

Reply:
This is a good suggestion. In the long run, it would be even better to make the essential boundary condition calls use `tensor`, which would make this code more compact and readable. It would also increase the uniformity of the interface with the other loading functions.